### ISSUE TITLE
Support installer windows tests

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/aqa-tests#4350]
Support installer windows tests

**Comment by smlambert** [comment]
Some notes from the chat that @sophia-guo and I just had (where we talked about how to treat these use cases in a standard way, with a goal to not keep adding an endless number of parameters).
- consider getting rid of all of the individual boolean parameters (TEST_IMAGES_REQUIRED,DEBUG_IMAGES_REQUIRED,USE_JRE,USE_INSTALLER,USE_RI_JDK) into a comma-separated string of 'constant' accepted values, perhaps called something like ADDITIONAL_ARTIFACTS_REQUIRED, which then could be set for all the additional things needed (over and above the JDK binary, which for simplicity and stated below, should always be passed in).
  - ADDITIONAL_ARTIFACTS_REQUIRED=TEST_IMAGES,DEBUG_IMAGES,JRE
  - ADDITIONAL_ARTIFACTS_REQUIRED=RI_JDK (only with SDK_RESOURCE=customized, since it will not be pulled from an upstream location or from the API as with nightly|releases)
  - ADDITIONAL_ARTIFACTS_REQUIRED=INSTALLER
- in all cases, the JDK is also wanted (in order to compile test material), so it should always be first in the list of items listed in CUSTOMIZED_SDK_URL when SDK_RESOURCE=customized

**Comment by Mesbah-Alam** [comment]
Note: Unlike Openj9 Mac binaries, we only get `.dmg` files for xMac and aMac RI SDKs, meaning we will require a different command to install Mac RI JDKs (for tck).

**Comment by smlambert** [comment]
Yes, for RI_JDK, the full path to the binary would be supplied via CUSTOMIZED_SDK_URL, so in the case of Mac JDKs, CUSTOMIZED_SDK_URL=https://someServer/pathTo/some.dmg and CUSTOMIZED_SDK_URL_CREDENTIAL_ID would be the ID required to login to someServer.

**Comment by smlambert** [comment]
For INSTALLER artifacts, we also look at PLATFORM, to know that if PLATFORM=x86-64_windows|x86-32_windows, then we'd be planning to pick up a .msi file.  For PLATFORM=someLinux platform, there are multiple types, .rpm/.deb, etc, so we will want to pass those in explicitly via the CUSTOMIZED_SDK_URL parameter.  (SDK_RESOURCE=upstream|nightly|releases for INSTALLER files can be out of scope for this issue).

**Comment by smlambert** [comment]
Thought of another use case for RI_JDK
- where we want to pass in 2 Temurin builds to compare them (and write tests related to reproducibility, including the simple one of 'diff' of the 2 builds)

**Comment by Mesbah-Alam** [comment]
Thanks @sophia-guo for assigning this issue to me. Since the multijvm work is blocked by this, it'd be good to get it implemented asap.

**Comment by smlambert** [comment]
Since RI_JDK is needed first, thanks @Mesbah-Alam for taking that on as the first of the 3 tasks (outlined in https://github.com/adoptium/aqa-tests/issues/4350#issuecomment-1447194197) and repeated below:

- ADDITIONAL_ARTIFACTS_REQUIRED=RI_JDK  

- ADDITIONAL_ARTIFACTS_REQUIRED=INSTALLER  

- ADDITIONAL_ARTIFACTS_REQUIRED=TEST_IMAGES,DEBUG_IMAGES,JRE 

We will add a table to this issue to outline the expected behaviour, depending on the parameters being used/set.

**Comment by smlambert** [comment]
Adding this table (WIP) for the purposes of outlining various scenarios and listing the expected behaviour or potential 'gotchas' for those scenarios.  This can later be added as part of detailed documentation reference.  Will keep fleshing this out, and noting that we may add 'constraints' around what is allowable to list in the ADDITIONAL_ARTIFACTS_REQUIRED at the same time for a particular run.

| ADDITIONAL_ARTIFACTS_REQUIRED | JDK_IMPL | SDK_RESOURCE | CUSTOMIZED_SDK_URL |Additional notes   | 
| --- | --- | --- | --- | --- | 
| RI_JDK | openj9 | nightly | 2 entries: TEST JDK, RI_JDK|Pull TEST JDK from artifactory, pull RI_JDK from server needing CUSTOMIZED_SDK_URL_CREDENTIAL_ID | 
| RI_JDK | openj9 | customized |2 entries: TEST JDK, RI_JDK | will 2 separate CUSTOMIZED_SDK_URL_CREDENTIAL_IDs ever be needed? For now, we say 'not supported' | 
| RI_JDK | openj9 | upstream | 1 entry: RI_JDK |  need to supply UPSTREAM_JOB_NAME & ID where TEST JDK is fetched from,  |
| RI_JDK | hotspot (temurin) | nightly  |  1 entry: RI_JDK |  Pull TEST JDK from Adoptium API, pull RI_JDK from CUSTOMIZED_SDK_URL |
| RI_JDK | hotspot (temurin) | customized  | 2 entries: TEST JDK, RI_JDK |  |
| RI_JDK | hotspot (temurin) | upstream  | 1 entry: RI_JDK| |need to supply UPSTREAM_JOB_NAME & ID |
| | | | |
| INSTALLER | openj9 | customized | 2 entries: TEST JDK, RI_JDK| | |
| INSTALLER | hotspot | customized | 2 entries: TEST JDK, RI_JDK| | |
| | | | |
| TEST_IMAGES,DEBUG_IMAGES | openj9 | customized | 2 entries: TEST JDK, RI_JDK| | |
| TEST_IMAGES,DEBUG_IMAGES | hotspot | customized | 2 entries: TEST JDK, RI_JDK| | |

**Comment by Mesbah-Alam** [comment]
RE: https://github.com/adoptium/aqa-tests/issues/4350#issuecomment-1453817503 

For the combination of openj9 + nightly (first row in the table), there is an issue - 'nightly' links are openj9 artifactory, which requires that NO credentials be supplied via CUSTOMIZED_SDK_URL_CREDENTIAL_ID. However, RI JDK will be pulled from internal artifactory - which will need the credentials supplied via CUSTOMIZED_SDK_URL_CREDENTIAL_ID. So, this combination may not work, at least for Openj9.

**Comment by smlambert** [comment]
Ya, let's focus on the SDK_RESOURCE=customized case, as it is likely most relevant.  :)
